src/app/templates/activity/ActivityItem/ActivityOperationBase/index.tsx

```diff
@@ -0,0 +1,228 @@
+
+      return (
+        <div
+          className={clsx(
+            'max-w-40 flex text-font-num-14 overflow-hidden',
+            asset.amountSigned && Number(asset.amountSigned) > 0 && 'text-success',
+            onClick && 'group-hover:hidden'
+          )}
+        >
+          {kind === ActivityOperKindEnum.approve ? null : asset.amountSigned ? (
+            <Money smallFractionFont={false} withSign>
+              {atomsToTokens(asset.amountSigned, asset.decimals)}
+            </Money>
+          ) : null}
+
+          {symbolStr ? <span className="whitespace-pre"> {symbolStr}</span> : null}
+        </div>
+      );
+    }, [asset, kind, onClick]);
+
+    const fiatJsx = useMemo<ReactNode>(() => {
+      if (!asset) return null;
+
+      if (!asset.amountSigned) return asset.amountSigned === null ? 'Unlimited' : null;
+
+      if (kind === ActivityOperKindEnum.approve)
+        return <Money smallFractionFont={false}>{atomsToTokens(asset.amountSigned, asset.decimals)}</Money>;
+
+      if (!assetSlug) return null;
+
+      const amountForFiat =
+        kind === 'bundle' || isTransferActivityOperKind(kind)
+          ? atomsToTokens(asset.amountSigned, asset.decimals)
+          : null;
+
+      if (!amountForFiat) return null;
+
+      return (
+        <InFiat
+          evm={typeof chainId === 'number'}
+          chainId={chainId}
+          assetSlug={assetSlug}
+          volume={amountForFiat}
+          smallFractionFont={false}
+          withSign
+        >
+          {({ balance, symbol, noPrice }) =>
+            noPrice ? (
+              <span>No value</span>
+            ) : (
+              <>
+                {balance}
+                <span className="ml-1">{symbol}</span>
+              </>
+            )
+          }
+        </InFiat>
+      );
+    }, [asset, kind, assetSlug, chainId]);
+
+    const chipJsx = useMemo(
+      () =>
+        addressChip ?? (
+          <Anchor
+            href={blockExplorerUrl}
+            target="_blank"
+            className="flex items-center gap-x-1 group-hover:text-secondary"
+          >
+            <span>
+              <HashShortView hash={hash} firstCharsCount={6} lastCharsCount={4} />
+            </span>
+
+            <IconBase Icon={OutLinkIcon} size={12} className="invisible group-hover:visible" />
+          </Anchor>
+        ),
+      [addressChip, hash, blockExplorerUrl]
+    );
+
+    const isNFT = Boolean(asset?.nft);
+
+    const faceIconJsx = useMemo(() => {
+      if (withoutAssetIcon || !assetSlug)
+        return (
+          <div className="w-full h-full flex items-center justify-center bg-grey-4">
+            <IconBase Icon={getIconByKind(kind, transferType)} className="text-grey-1" />
+          </div>
+        );
+
+      const className = 'w-full h-full object-cover';
+
+      const placeholderJsx = <AssetIconPlaceholder isNFT={isNFT} symbol={asset?.symbol} className={className} />;
+
+      return typeof chainId === 'number' ? (
```

Comment:
i thought we agreed on chainId type check as a bad practice, pls add prop `isEvm` or just pass the whole chain and check its chainKind